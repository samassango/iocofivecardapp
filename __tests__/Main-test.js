import 'react-native';
import React from 'react';
import Carousel from 'react-native-snap-carousel';
import { Text } from 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Main from '../src/components/main';
import NavigationIcon from '../src/components/navigationicon';
var carouselRef;

it('renders main coimponent correctly', () => {
    const mainTree = renderer.create(<Main />);
    expect(mainTree).toMatchSnapshot();
});

it('renders the carousel correctly', () => {
    var carouselRef;
    const carousel = renderer.create(<Carousel
        layout={"default"}
        ref={ref => carouselRef = ref}
        data={[{ title: 'item 1' }, { title: 'item 2' }, { title: 'item 3' }]}
        sliderWidth={200}
        itemWidth={200}
        renderItem={({ item, index }) => <Text>{item.title}</Text>}
    />).toJSON();

    expect(carousel).toMatchSnapshot();
});
const mockFn = jest.fn();
const navIconTree = renderer.create(<NavigationIcon icon={"navigate-before"} onEventTouch={mockFn} size={50} color={"#fff"} />);

it('renders navigationicon component correctly', () => {
    expect(navIconTree).toMatchSnapshot();
});

const carouselTree = renderer.create(<Carousel
    layout={"default"}
    ref={ref => carouselRef = ref}
    data={[{ title: 'item 1' }, { title: 'item 2' }, { title: 'item 3' }]}
    sliderWidth={200}
    itemWidth={200}
    renderItem={({ item, index }) => <Text>{item.title}</Text>}
/>);

const mock = jest.fn().mockImplementation(() => {
    return [{ title: 'item 1' }, { title: 'item 2' }, { title: 'item 3' }];
});

it('renders correct layout prop', () => {
    const testInstance = carouselTree.root;
    expect(testInstance.props.layout).toBe("default");
})

it('renders correct data', () => {
    const testInstance = carouselTree.root;
    expect(testInstance.props.data).toEqual(mock());
})

it('renders correct slider width size', () => {
    const testInstance = carouselTree.root;
    expect(testInstance.props.sliderWidth).toEqual(200);
})

it('renders correct item width size', () => {
    const testInstance = carouselTree.root;
    expect(testInstance.props.itemWidth).toEqual(200);
})