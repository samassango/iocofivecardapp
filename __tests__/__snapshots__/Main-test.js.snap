// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main coimponent correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#42403a",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 667,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 667,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="arrow-navigate-before"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Óêà
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "justifyContent": "center",
        }
      }
    >
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
        data={
          Array [
            Object {
              "src": Object {
                "testUri": "../../../src/assets/1.png",
              },
              "text": "START HERE",
              "title": "Get a device",
            },
            Object {
              "src": Object {
                "testUri": "../../../src/assets/2.png",
              },
              "text": "START HERE",
              "title": "Mobile internet",
            },
            Object {
              "src": Object {
                "testUri": "../../../src/assets/3.png",
              },
              "text": "START HERE",
              "title": "Home internet",
            },
            Object {
              "src": Object {
                "testUri": "../../../src/assets/4.png",
              },
              "text": "START HERE",
              "title": "Add a phone-line",
            },
            Object {
              "src": Object {
                "testUri": "../../../src/assets/5.png",
              },
              "text": "START HERE",
              "title": "Upgrade",
            },
          ]
        }
        decelerationRate="fast"
        directionalLockEnabled={true}
        enableMomentum={false}
        enableSnap={true}
        firstItem={0}
        forwardedRef={[Function]}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={6}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={750}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={13}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={750}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": 750,
          }
        }
        swipeThreshold={20}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={13}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#322e37",
                    "borderBottomColor": "#f5c911",
                    "borderBottomWidth": 5,
                    "borderRadius": 10,
                    "borderWidth": 0,
                    "elevation": 20,
                    "height": 667,
                    "padding": 20,
                    "shadowColor": "#f5c911",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 3,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                    "width": 280,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "padding": 20,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/assets/1.png",
                      }
                    }
                    style={
                      Object {
                        "height": 130,
                        "width": 130,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 30,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Get a device
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 60,
                      "position": "absolute",
                      "width": 250,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "margin": 10,
                        "opacity": 1,
                        "paddingLeft": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 15,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      START HERE
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Óóå
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#322e37",
                    "borderBottomColor": "#f5c911",
                    "borderBottomWidth": 5,
                    "borderRadius": 10,
                    "borderWidth": 0,
                    "elevation": 20,
                    "height": 667,
                    "padding": 20,
                    "shadowColor": "#f5c911",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 3,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                    "width": 280,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "padding": 20,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/assets/2.png",
                      }
                    }
                    style={
                      Object {
                        "height": 130,
                        "width": 130,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 30,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Mobile internet
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 60,
                      "position": "absolute",
                      "width": 250,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "margin": 10,
                        "opacity": 1,
                        "paddingLeft": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 15,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      START HERE
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Óóå
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#322e37",
                    "borderBottomColor": "#f5c911",
                    "borderBottomWidth": 5,
                    "borderRadius": 10,
                    "borderWidth": 0,
                    "elevation": 20,
                    "height": 667,
                    "padding": 20,
                    "shadowColor": "#f5c911",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 3,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                    "width": 280,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "padding": 20,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/assets/3.png",
                      }
                    }
                    style={
                      Object {
                        "height": 130,
                        "width": 130,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 30,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Home internet
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 60,
                      "position": "absolute",
                      "width": 250,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "margin": 10,
                        "opacity": 1,
                        "paddingLeft": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 15,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      START HERE
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Óóå
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#322e37",
                    "borderBottomColor": "#f5c911",
                    "borderBottomWidth": 5,
                    "borderRadius": 10,
                    "borderWidth": 0,
                    "elevation": 20,
                    "height": 667,
                    "padding": 20,
                    "shadowColor": "#f5c911",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 3,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                    "width": 280,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "padding": 20,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/assets/4.png",
                      }
                    }
                    style={
                      Object {
                        "height": 130,
                        "width": 130,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 30,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Add a phone-line
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 60,
                      "position": "absolute",
                      "width": 250,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "margin": 10,
                        "opacity": 1,
                        "paddingLeft": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 15,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      START HERE
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Óóå
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#322e37",
                    "borderBottomColor": "#f5c911",
                    "borderBottomWidth": 5,
                    "borderRadius": 10,
                    "borderWidth": 0,
                    "elevation": 20,
                    "height": 667,
                    "padding": 20,
                    "shadowColor": "#f5c911",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 3,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                    "width": 280,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "padding": 20,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/assets/5.png",
                      }
                    }
                    style={
                      Object {
                        "height": 130,
                        "width": 130,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 30,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Upgrade
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 60,
                      "position": "absolute",
                      "width": 250,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "margin": 10,
                        "opacity": 1,
                        "paddingLeft": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 15,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      START HERE
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Óóå
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 667,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="arrow-navigate-next"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Óêâ
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders navigationicon component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 667,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="arrow-navigate-before"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    Óêà
  </Text>
</View>
`;

exports[`renders the carousel correctly 1`] = `
<RCTScrollView
  activeAnimationOptions={null}
  activeAnimationType="timing"
  activeSlideAlignment="center"
  activeSlideOffset={20}
  apparitionDelay={0}
  automaticallyAdjustContentInsets={false}
  autoplay={false}
  autoplayDelay={1000}
  autoplayInterval={3000}
  callbackOffsetMargin={5}
  containerCustomStyle={Object {}}
  contentContainerCustomStyle={Object {}}
  contentContainerStyle={
    Array [
      Object {
        "paddingLeft": 0,
        "paddingRight": 0,
      },
      Object {},
    ]
  }
  data={
    Array [
      Object {
        "title": "item 1",
      },
      Object {
        "title": "item 2",
      },
      Object {
        "title": "item 3",
      },
    ]
  }
  decelerationRate="fast"
  directionalLockEnabled={true}
  enableMomentum={false}
  enableSnap={true}
  firstItem={0}
  forwardedRef={[Function]}
  getItem={[Function]}
  getItemCount={[Function]}
  hasParallaxImages={false}
  horizontal={true}
  inactiveSlideOpacity={0.7}
  inactiveSlideScale={0.9}
  inactiveSlideShift={0}
  initialNumToRender={6}
  invertStickyHeaders={false}
  inverted={false}
  itemWidth={200}
  keyExtractor={[Function]}
  layout="default"
  lockScrollTimeoutDuration={1000}
  lockScrollWhileSnapping={false}
  loop={false}
  loopClonesPerSide={3}
  maxToRenderPerBatch={13}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderRelease={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  overScrollMode="never"
  pinchGestureEnabled={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={1}
  scrollsToTop={false}
  shouldOptimizeUpdates={true}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  slideStyle={Object {}}
  sliderWidth={200}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 0,
      "width": 200,
    }
  }
  swipeThreshold={20}
  useScrollView={false}
  vertical={false}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={13}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        pointerEvents="box-none"
        style={
          Object {
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 200,
          }
        }
      >
        <Text>
          item 1
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        pointerEvents="box-none"
        style={
          Object {
            "opacity": 0.7,
            "transform": Array [
              Object {
                "scale": 0.9,
              },
            ],
            "width": 200,
          }
        }
      >
        <Text>
          item 2
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        pointerEvents="box-none"
        style={
          Object {
            "opacity": 0.7,
            "transform": Array [
              Object {
                "scale": 0.9,
              },
            ],
            "width": 200,
          }
        }
      >
        <Text>
          item 3
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
